src/query/query_utils.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{
+    error::ContractError,
+    models::PublicNameRecord,
+    state::{NAME_METADATA, NAME_RECORDS},
+};
+use super::ReadonlyContext;
+
+pub fn build_public_name_record(ctx: &ReadonlyContext, cannonical_name: String) -> Result<PublicNameRecord, ContractError> {
```

Comment:
Could possibly make this into a method on the NameRecord struct

Reply:
done